uaclient/config.py

```diff
@@ -526,6 +526,18 @@ def status(self, show_beta=False) -> "Dict[str, Any]":
 
         Write the status-cache when called by root.
         """
+
+        # Try to remove all reboot notices if not applicable
+        if not util.should_reboot():
+            self.remove_notice("", status.MESSAGE_FIPS_REBOOT_REQUIRED)
+            self.remove_notice("", status.MESSAGE_FIPS_DISABLE_REBOOT_REQUIRED)
```

Comment:
I think we are missing status.MESSAGE_LIVEPATCH_LTS_REBOOT_REQUIRED too and I think this also means we can drop the remove_notices logic from reboot_cmds.py too

Reply:
For the `FIPS entitlements` case this will not happen because we check in `ua status`  the existence of the `/var/run/reboot-required` which indeed exists as can be seen here:
https://github.com/canonical/ubuntu-advantage-client/blob/master/uaclient/entitlements/fips.py#L56

Now, for the FIPS PRO case, yeah, this will happen. But we can avoid that by also checking if the marker file is not present.
In that case, we will guarantee that the FIPS PRO case will be handled by the reboot script only and the rest of the cases will be handled by status.

What do you think ?